components/common/src/command/package/download.rs

```diff
@@ -0,0 +1,400 @@
+//! Downloads a Habitat package from a [depot](../depot).
+//!
+//! # Examples
+//!
+//! ```bash
+//! $ hab pkg download core/redis
+//! ```
+//!
+//! Will download `core/redis` package from a custom depot:
+//!
+//! ```bash
+//! $ hab pkg download core/redis/3.0.1 redis -u http://depot.co:9633
+//! ```
+//!
+//! This would download the `3.0.1` version of redis.
```

Comment:
Downloads the latest release for the 3.0.1 version..
Along with tdeps and public keys...
(for the specified target, into the specified folder in a specific format..)